src/test/test-fileio.c

```diff
@@ -427,15 +427,15 @@ static void test_write_string_file(void) {
 static void test_write_string_file_no_create(void) {
         _cleanup_(unlink_tempfilep) char fn[] = "/tmp/test-write_string_file_no_create-XXXXXX";
         _cleanup_close_ int fd;
-        char buf[64] = {0};
+        char buf[64] = {};
 
         fd = mkostemp_safe(fn);
         assert_se(fd >= 0);
 
         assert_se(write_string_file("/a/file/which/does/not/exists/i/guess", "boohoo", 0) < 0);
         assert_se(write_string_file(fn, "boohoo", 0) == 0);
 
-        assert_se(read(fd, buf, sizeof(buf)) == STRLEN("boohoo\n"));
+        assert_se((size_t) read(fd, buf, sizeof buf) == strlen("boohoo\n"));
```

Comment:
not that it matters, but I'd probably have cast the strle() to ssize_t rather than the read() to size_t, since we know for sure the result of this strlen() will fit into the ssize_t range, but the read() return value does not fit into size_t on read error...